### PR TITLE
e2e test to include bookinfo as test application

### LINKED ISSUE #766: [E2E Enhancement] include bookinfo as test appliaction
**What would you like to be added**:

Include [bookinfo](https://istio.io/latest/docs/examples/bookinfo/) as test application in the e2e test set to cover complex application scenarios with multiple language applications and multiple hops. In this way, problem similar to #756 could be convered.

**Why is this needed**:

The test application in the e2e framework is just a multi-instance deployment of a single go application, which acts as both a client and a server, and has only a single-hop access. When doing manual testing, we often use bookinfo, but it often happens that all e2e test cases pass and the bookinfo test fails.

### PR DESCRIPTION
**Pull request by YaoZengzeng**
**What type of PR is this?**

/kind enhancement

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #766

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: test/e2e/baseline_test.go [hunk 1]
```diff
@@ -610,6 +612,77 @@ func TestMixNsAndServiceWaypoint(t *testing.T) {
...
 
+func TestBookinfo(t *testing.T) {
+	framework.NewTest(t).Run(func(t framework.TestContext) {
+		namespace := apps.Namespace.Name()
+		// Install bookinfo.
+		if _, err := shell.Execute(true, fmt.Sprintf("kubectl apply -f https://raw.githubusercontent.com/istio/istio/release-1.22/samples/bookinfo/platform/kube/bookinfo.yaml -n %s", namespace)); err != nil {
+			t.Fatalf("failed to install bookinfo: %v", err)
+		}
+		t.Cleanup(func() {
+			if _, err := shell.Execute(true, fmt.Sprintf("kubectl delete -f https://raw.githubusercontent.com/istio/istio/release-1.22/samples/bookinfo/platform/kube/bookinfo.yaml -n %s", namespace)); err != nil {
```
Comment (LiZhenCheng9527) [inline test/e2e/baseline_test.go:623]: Does this need to change with istio version changes

File: test/e2e/baseline_test.go [hunk 2]
```diff
@@ -610,6 +612,77 @@ func TestMixNsAndServiceWaypoint(t *testing.T) {
...
+				t.Fatalf("failed to delete bookinfo: %v", err)
+			}
+		})
+
+		// Install sleep as client.
+		if _, err := shell.Execute(true, fmt.Sprintf("kubectl apply -f https://raw.githubusercontent.com/istio/istio/release-1.21/samples/sleep/sleep.yaml -n %s", namespace)); err != nil {
```
Comment (LiZhenCheng9527) [inline test/e2e/baseline_test.go:629]: Why is it release 1.22 above and release 1.21 here?

File: test/e2e/baseline_test.go (hunk 1 above)
Comment (YaoZengzeng) [reply test/e2e/baseline_test.go:623]: > Does this need to change with istio version changes

It should not be needed at present. We only use very basic functions here and bookinfo itself will not change much with upgrade of istio.

**Review by hzxuzhonghu — APPROVED** [review]
This is very hack,  we can use call options provided by istio


But i wish this test to cover ipv4 mapped v6 asap, so

/lgtm

**Pull request merged**